src/api/useTestCaseServiceApi.ts

```diff
@@ -0,0 +1,40 @@
+import axios, { AxiosResponse } from "axios";
+import useServiceConfig from "./useServiceConfig";
+import { ServiceConfig } from "./ServiceContext";
+import {
+  HapiOperationOutcome,
+  TestCase,
+  TestCaseImportRequest,
+} from "@madie/madie-models";
+import { useOktaTokens } from "@madie/madie-util";
+
+export class TestCaseServiceApi {
+  constructor(private baseUrl: string, private getAccessToken: () => string) {}
+
+  async getTestCasesByMeasureId(measureId: string): Promise<TestCase[]> {
+    try {
+      const response = await axios.get<TestCase[]>(
+        `${this.baseUrl}/measures/${measureId}/test-cases`,
+        {
+          headers: {
+            Authorization: `Bearer ${this.getAccessToken()}`,
+          },
+        }
+      );
+      return response.data || [];
+    } catch (err) {
+      const message = "Unable to retrieve test cases, please try later.";
+      throw new Error(message);
```

Comment:
It will be nice to see a unit test suite for the new class

Reply:
Also seems like we are generalizing the error message displayed to the user.
The catch block will be executed when axios received a non-2xx response, So for example if the server responds with 401 we are still asking user to "please try later", but we are unable to retrieve test cases because user doesn't have permissions.